### ISSUE TITLE
recursive class init

### ISSUE THREAD

**Issue by forestchen1224** [PyCQA/pylint#3100]
### Steps to reproduce
```python
class XmlDictConfig(dict):
    '''
    Example usage:

    >>> tree = ElementTree.parse('your_file.xml')
    >>> root = tree.getroot()
    >>> xmldict = XmlDictConfig(root)

    Or, if you want to use an XML string:

    >>> root = ElementTree.XML(xml_string)
    >>> xmldict = XmlDictConfig(root)

    And then use xmldict for what it is... a dict.
    '''
    def __init__(self, parent_element):
        if parent_element.items():
            self.update(dict(parent_element.items()))
        for element in parent_element:
            if element:
                if len(element) == 1 or element[0].tag != element[1].tag:
                    aDict = XmlDictConfig(element)
                else:
                    aDict = {element[0].tag: XmlListConfig(element)}
                if element.items():
                    aDict.update(dict(element.items()))
                self.update({element.tag: aDict})

            elif element.items():
                self.update({element.tag: dict(element.items())})
            else:
                self.update({element.tag: element.text})
  ```
recursive init a class XmlListConfig, the linter report undefined variable
### Current behavior
Undefined variable 'XmlListConfig'

### pylint --version output
Version: 2.3.1

**Comment by PCManticore** [comment]
Well `pylint` is correct here as the class is named `XmlDictConfig` and there is no `XmlListConfig` class in this small snippet.

**Comment by forestchen1224** [comment]
But there is a error notification and the code is right.  
so pylint can't slove these recursive problem?
